### ISSUE TITLE
Workspaces in Reports don't update

### ISSUE THREAD

**Issue by amulware** [toggl/mobileapp#3071]
## 🐛 Bug Report
The workspace collection of the reports view model is initialised only once on startup and does not update when the workspaces change.

## Context
This is a follow up to #3070.

## Expected behavior
The workspace list and button state should be up to date.

## To reproduce
- be logged in
- create a new workspace on the web
- sync with the app, you can use the workspace for TEs now
- go to reports
- observe lack of the new workspace

## Suggestions
I propose to update the workspaces in the reports view model, as well as the dependent properties, whenever the reports view is opened. That's good enough in my opinion.

**Comment by amulware** [comment]
Could maybe be done together with #3072

**Comment by amulware** [comment]
Duplicate of #2982
